layers/+window-management/spacemacs-layouts/funcs.el

```diff
@@ -117,6 +122,25 @@ perspectives does."
                       (projectile-switch-project-by-name project)))))))
    :buffer "*Helm Projectile Layouts*"))
 
+(defun spacemacs/helm-persp-switch-open-project (arg)
+  (interactive "P")
+  (helm
+   :sources
+   (helm-build-in-buffer-source "*Helm Switch Project Layout*"
+     :data (lambda ()
+             (if (projectile-project-p)
+                 layouts-projectile
+               projectile-known-projects))
```

Comment:
Also, switching to a known project with this function makes Emacs freeze. I don't know yet how to debug that. :confused: